rust/src/sip/sip.rs

```diff
@@ -518,6 +518,59 @@ pub unsafe extern "C" fn rs_sip_parse_response_tcp(
     state.parse_response_tcp(flow, stream_slice)
 }
 
+fn register_pattern_probe(proto: u8) -> i8 {
+    let methods: Vec<&str> = vec![
+        "REGISTER\0",
```

Comment:
I do not think you need the final 0 in rust, do you ?

Reply:
It seems that the same logic is applied in `bittorrent_dht.rs`

```
const BITTORRENT_DHT_PAYLOAD_PREFIX: &[u8] = b"d1:ad2:id20:\0";

if AppLayerProtoDetectPMRegisterPatternCS(
    IPPROTO_UDP,
    ALPROTO_BITTORRENT_DHT,
    BITTORRENT_DHT_PAYLOAD_PREFIX.as_ptr() as *const c_char,
    BITTORRENT_DHT_PAYLOAD_PREFIX.len() as u16 - 1,
    0,
    crate::core::Direction::ToServer.into(),
) < 0
{
    SCLogDebug!("Failed to register protocol detection pattern for direction TOSERVER");
};
```

Reply:
https://redmine.openinfosecfoundation.org/issues/6583 ;-)